h2/src/main/org/h2/engine/MetaRecord.java

```diff
@@ -24,6 +24,13 @@
     private final int objectType;
     private final String sql;
 
+    public static void populateRowFromDBObject(DbObject obj, SearchRow r) {
```

Comment:
It looks like this new static method can be used in `Database.addMeta()` too and `MetaRecord.setRecord()` can be removed to avoid code duplication that was introduced here.